org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java

```diff
@@ -398,12 +417,23 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
      * @param in input stream
      * @throws IOException
      */
-    static void writeToKnimeConsole(final InputStream in) throws IOException {
-        BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
-        String line;
-        while ((line = bis.readLine()) != null) {
-            KNIPGateway.log().info(line);
-        }
+    static void redirectToKnimeConsole(final InputStream in) {
+
+        new Thread(
+                   new Runnable() {
+                       @Override
+                       public void run() {
+                           BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
+                           String line;
+                           try {
+                               while ((line = bis.readLine()) != null) {
+                                   KNIPGateway.log().debug(line);
+                               }
+                           } catch (IOException ioe) {
+                               throw new RuntimeException("Could not read ilastik output", ioe); //FIXME not sure if this is a good idea
```

Comment:
It will probably not even propagate outside of the thread.

http://stackoverflow.com/questions/6546193/how-to-catch-an-exception-from-a-thread

And in this case, it's not really problematic if an error occurs, right? All you're trying to do is forward the output of the process into the KNIME debug log and that's not really cruical. (I would probably even ignore the IOException)